src/SfxWeb/src/app/Models/DataModels/Replica.ts

```diff
@@ -73,13 +73,31 @@ export class ReplicaOnPartition extends DataModelBase<IRawReplicaOnPartition> {
     }
 
     public get role(): string {
+        const previousRole: string = this.raw.PreviousReplicaRole ? `Previous Role: ${this.raw.PreviousReplicaRole} | ` : '';
+
+        if (this.parent.raw.PartitionStatus === 'Reconfiguring') {
+            return `Reconfiguring - ${previousRole}Target Role: ${this.raw.ReplicaRole}`;
+        }
+
+        return this.raw.ReplicaRole;
+    }
+
+    public get currentRole(): string {
         if (this.parent.raw.PartitionStatus === 'Reconfiguring') {
             return `Reconfiguring - Target Role: ${this.raw.ReplicaRole}`;
         }
 
         return this.raw.ReplicaRole;
     }
 
+    public get previousRole(): string {
```

Comment:
Shouldnt this always return previous replica role unless its a new replica?

Reply:
Could potential compatibility problems arise if the customer uses the most recent SFX with an SF version that predates this addition of the `PreviousReplicaRole` to the [Get-ServiceFabricReplica
](https://learn.microsoft.com/en-us/powershell/module/servicefabric/get-servicefabricreplica?view=azureservicefabricps) API? I'm uncertain about the testing procedure for this scenario, or whether it's necessary at all.

Reply:
If there are no issues with backwards compatibility, then we can completely remove this if check and always return `PreviousReplicaRole`.

Reply:
We cant necessarily ensure that. So it is good to have the backwards compat in place.